engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/number/integer/LessNode.java

```diff
@@ -9,15 +9,30 @@
 import org.enso.interpreter.runtime.error.DataflowError;
 import org.enso.interpreter.runtime.number.EnsoBigInteger;
 
-@BuiltinMethod(type = "Big_Integer", name = "<", description = "Comparison of numbers.")
+@BuiltinMethod(type = "Integer", name = "<", description = "Comparison of numbers.")
 public abstract class LessNode extends Node {
 
-  abstract Object execute(EnsoBigInteger self, Object that);
+  abstract Object execute(Object self, Object that);
 
   static LessNode build() {
     return LessNodeGen.create();
   }
 
+  @Specialization
+  boolean doLong(long self, long that) {
+    return self < that;
+  }
+
+  @Specialization
+  boolean doDouble(long self, double that) {
+    return (double) self < that;
+  }
+
+  @Specialization
+  boolean doBigInteger(long self, EnsoBigInteger that) {
+    return that.getValue().signum() > 0;
+  }
+
   @Specialization
   boolean doDouble(EnsoBigInteger self, double that) {
     return BigIntegerOps.toDouble(self.getValue()) < that;
```

Comment:
Shall we rename these to avoid overloads?

I'm not sure if this has any implications for the compilation, probably no.

But still, it feels cleaner and more consistent to have names be unique - AFAIK in other places all of our specializations do have unique names.